### ISSUE TITLE
Question about reproducing bge-en-icl

### ISSUE THREAD

**Issue by greeneggsandyaml** [FlagOpen/FlagEmbedding#1158]
Hello authors, thank you for your great work! 

I am trying to reproduce your `bge-en-icl` results. I saw your example command in the `README.md` file, but it looks like this might be a toy run. Would you be able to post the command used to train the actual model? 

I'm particularly interested in the max query length. The paper says "the maximum sequence length remains capped at 512 tokens", and the example command in the repository has `--query_max_len 512`. However, there is the following logic in `data.py`: https://github.com/FlagOpen/FlagEmbedding/blob/dc2170d46fab2ecb5631317dc1924aae0229c84a/FlagEmbedding/llm_dense_retriever/finetune/data.py#L284-L305

If I understand correctly, when `self.query_max_len` is 512, then `self.query_max_len - 512` will always be 0, so the `break` will always be triggered, which means that no icl pairs will ever be added to the prompt. 

I've run the code with the example settings and this is the result I observe; there are no in-context examples.

Thanks for all your help and for your papers!

**Comment by 545999961** [comment]
Hello, I apologize for the misunderstanding regarding the parameters. You can use the following parameters:

```
--query_max_len 512 \
--passage_max_len 512 \
--example_query_max_len 256 \
--example_passage_max_len 256
```

The technical report states that "The maximum allowable lengths for example queries and documents are set to 256 tokens each, and the combined length for a query with examples is set at 2048 tokens." I apologize for incorrectly presenting a `query_max_len` of 512 in the README. We will revise this section. Thank you very much for bringing this issue to our attention.

**Comment by greeneggsandyaml** [comment]
Thanks, I appreciate the quick response! 

A second question, this time about the batch size: how did you train with a batch size of 512 (which is what I see in the paper)? I'm asking because I have a single 8xH100 node and I don't think I can fit that batch size. Did you use multiple nodes / gradient accumulation / something else?  

Thanks for all the help!

**Comment by 545999961** [comment]
> Thanks, I appreciate the quick response!
> 
> A second question, this time about the batch size: how did you train with a batch size of 512 (which is what I see in the paper)? I'm asking because I have a single 8xH100 node and I don't think I can fit that batch size. Did you use multiple nodes / gradient accumulation / something else?
> 
> Thanks for all the help!

We use two machines with 8×A800 GPUs each for training.

**Comment by greeneggsandyaml** [comment]
Thanks, great. I'm closing issue -- I'll create a new one if I have any more questions!
